lib/recommendation.js

```diff
@@ -0,0 +1,61 @@
+var XMLHttpRequest = Cc['@mozilla.org/xmlextras/xmlhttprequest;1'].createInstance();
```

Comment:
Alas, one more line note here: we should lazily instantiate the xhr, because FF has problems with slow initial loading